src/sentry/notifications/slack.py

```diff
@@ -0,0 +1,87 @@
+import logging
+from typing import Any, Iterable, Mapping, Optional
+
+
+from sentry.constants import ObjectStatus
+from sentry.integrations.slack.client import SlackClient
+from sentry.models import Integration
+from sentry.models.integration import ExternalProviders, ExternalUser, get_provider_name
+from sentry.notifications.constants import SLACK_TIMEOUT
+from sentry.shared_integrations.exceptions import ApiError
+from sentry.utils import json
+
+logger = logging.getLogger("sentry.notifications")
+
+
+def get_slack_channel_id_for_user(organization: Any, integration: Any, user: Any) -> Optional[str]:
```

Comment:
what do you think about moving this logic to the slack directory? maybe they could go in utils?